src/InstallationValidator.Core/Services/BatchComparisonReportingTask.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using InstallationValidator.Core.Assets;
+using InstallationValidator.Core.Domain;
+using OSPSuite.Core;
+using OSPSuite.Core.Reporting;
+using OSPSuite.Core.Services;
+using OSPSuite.Utility;
+
+namespace InstallationValidator.Core.Services
+{
+   public interface IBatchComparisonReportingTask
+   {
+      Task CreateReport(BatchComparisonResult comparisonResult, string firstFolderPath, string secondFolderPath, bool openReport);
+   }
+
+   public class BatchComparisonReportingTask : IBatchComparisonReportingTask
```

Comment:
That looks a lot like the other Task. I don't think we need to duplicate that code here. Let's create another method in the task that will compose some sub private method. EVerything is the same really except the argument of the CreateReport method.